tmt/steps/report/junit.py

```diff
@@ -58,40 +67,230 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+class ResultWrapper:
+    """
+    Adds possibility to wrap the ``tmt.Result`` and dynamically add more attributes which get
+    available inside the template context.
+    """
+
+    class PropertyDict(TypedDict):
+        """ Defines a property dict, which gets propagated into the final template properties. """
+
+        name: str
+        value: str
+
+    def __init__(self, wrapped: tmt.Result) -> None:
+        self._wrapped = wrapped
+        self._properties: dict[str, str] = {}
+
+    @property
+    def properties(self) -> list[PropertyDict]:
+        return [{'name': k, 'value': v} for k, v in self._properties.items()]
+
+    def set_properties(self, keyval: dict[str, str]) -> None:
+        self._properties = keyval
+
+    def __getattr__(self, name: str) -> Any:
+        """ Returns an attribute of a wrapped ``tmt.Result`` instance """
+        return getattr(self._wrapped, name)
+
+
+def _add_result_properties(result_instance: tmt.Result) -> ResultWrapper:
+    """ Dynamically decorate the ``tmt.Result`` instance - add ``properties`` attribute """
+    return ResultWrapper(result_instance)
+
+
+class ResultsContext:
+    """
+    A class which keeps the results context (especially the result summary) for
+    JUnit template. It wraps all the ``tmt.Result`` instances into the ``ResultWrapper``.
```

Comment:
The first line should bear a short summary, more details should be given on following lines. Also, use `:py:class:` when referring to other classes, Sphinx will turn it into nice links.

Reply:
I've added `:py:class:` and created short summaries. I hope it's resolved.